pkg/controllers/cluster/metrics.go

```diff
@@ -0,0 +1,49 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package cluster
+
+import (
+	"time"
+
+	greenhousev1alpha1 "github.com/cloudoperators/greenhouse/pkg/apis/greenhouse/v1alpha1"
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var (
+	kubernetesVersionsCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "greenhouse_cluster_k8s_versions_total",
+		},
+		[]string{"cluster", "namespace", "version"})
+
+	secondsToTokenExpiryGauge = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "greenhouse_cluster_kubeconfig_validity_seconds",
+		},
+		[]string{"cluster", "namespace"})
+)
+
+func init() {
+	metrics.Registry.MustRegister(kubernetesVersionsCounter)
+	metrics.Registry.MustRegister(secondsToTokenExpiryGauge)
+}
+
+//+kubebuilder:rbac:groups=greenhouse.sap,resources=clusters,verbs=get;list
+
+func UpdateMetrics(cluster *greenhousev1alpha1.Cluster) {
```

Comment:
Function should only be used by the Cluster controllers, no need to expose it to other packages.
```suggestion
func updateMetrics(cluster *greenhousev1alpha1.Cluster) {
```

Reply:
Done